doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -0,0 +1,602 @@
+additional authenticated data (I<aad>) and emits the ciphertext (I<ct>) to be
+sent to the recipient, alongside the encapsulated public value (I<enc>) output from
+OSSL_HPKE_encap(). 
+
+=head2 Recipient APIs
+
+Recipients using HPKE require a typically less ephemeral private value so that
+the public value can be distirbuted to potential senders via whatever protocol
+is using HPKE. For this reason, recipients will generally first generate a key
+pair and will need to manage their private key value using standard mechanisms
+outside the scope of this API. Private keys use normal L<EVP_PKEY(3)> pointers
+so normal private key management mechanisms can be used for the relevant
+values.
+
+In order to enable encapsulation, the recipient needs to make it's public value
+available to the sender. There is no generic HPKE format defined for that - the
+relevant formatting is intended to be defined by the application/protocols that
+makes use of HPKE. ECH for example defines an ECHConfig data structure that
+combines the public value with other ECH data items. Normal library functions
+must therefore be used to extract the public value in the required format based
+on the L<EVP_PKEY(3)> for the private value.
+
+OSSL_HPKE_keygen() provides a way for recipients to generate a key pair based
+on the HPKE I<suite> to be used. The main output is the L<EVP_PKEY(3)> pointer
+for the private value, I<priv>. I<libctx> and a property string (I<propq>) can
+be provided as optional inputs. 
+
+The HPKE specification also defines a deterministic key generation scheme where
+the private value is derived from initial keying material (IKM), so
+OSSL_HPKE_keygen() also has an option to use that scheme, using the I<ikm>
+parameter. When NULL/0 values are provided instead of a real IKM, then a
+randomly generated key for the relevant I<suite> will be produced.
+
+OSSL_HPKE_decap() takes as input the sender's encapsulated public value
+produced by OSSL_HPKE_encap() (I<enc>) and the recipient's L<EVP_PKEY(3)>
+pointer (I<prov>), and then re-generates the internal secret derived by the
+sender. As before, an optional I<info> parameter allows binding that derived
+secret to other application/protocol artefacts.
+
+OSSL_HPKE_open() is used by the recipient to recover an
+encrypted value sent to it, again with optional accompanying
+I<aad>. As with OSSL_HPKE_seal() there is an internally 
+incrementing nonce value so ciphertexts need to be presented
+in the correct order.
+
+=head2 Exporting Secrets
+
+HPKE defines a way to produce exported secrets for use by the
+appclication. For this to work, the sender must have frst called
+OSSL_HPKE_encap() and, in order to be able to derive the same exported secret,
+and the receiver must have called the corresponding OSSL_HPKE_decap().  
+The I<suite> used when forming an OSSL_HPKE_CTX to be used for exporting secrets
+must use the "fake" encryption algorithm denoted by the identifier
```

Comment:
I think you are doing something different from the RFC here if this is true.. It should work for both export only and any of the other modes..

Reply:
My text was wrong - export can be used after/with _seal(). I think I've fixed that text now.

Turns out I also had other wrong text here - the export is only based on the exporter_secret which is set after _encap() so there's no increment either. I added a test for that being the case to hpke_test.c/hpke_test_export() and fixed the text.

Reply:
think text and associated text should be ok now - also check code vs. RFC9180 and figure we're good